script/cms/compare-json.js

```diff
@@ -0,0 +1,340 @@
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+const deepDiff = require('deep-diff');
+const { map, camelCase } = require('lodash');
+const commandLineArgs = require('command-line-args');
+const commandLineUsage = require('command-line-usage');
+const assembleEntityTreeFactory = require('../../src/site/stages/build/process-cms-exports');
+const {
+  readEntity,
+} = require('../../src/site/stages/build/process-cms-exports/helpers');
+
+const commandLineDefs = [
+  {
+    name: 'entity',
+    alias: 'e',
+    type: String,
+    multiple: true,
+    description: 'Specify the specific node entity to compare.',
+  },
+  {
+    name: 'bundle',
+    type: String,
+    description:
+      'Compare all entities of the bundle type specified. (Only node entities are supported) i.e. node-q_a, node-health_care_local_facility, etc',
+    multiple: true,
+  },
+  {
+    name: 'help',
+    type: Boolean,
+    description: 'Show this help menu.',
+  },
+  {
+    name: 'buildtype',
+    type: String,
+    description: 'The buildtype to test the data against.',
+    defaultValue: 'localhost',
+  },
+];
+
+const { entity: entityNames, bundle, help, buildtype } = commandLineArgs(
+  commandLineDefs,
+);
+
+const exportDir = path.resolve(
+  __dirname,
+  `../../.cache/${buildtype}/cms-export-content`,
+);
+
+// Array of nodes from pages.json
+const graphQL = JSON.parse(
+  fs.readFileSync(
+    path.join(__dirname, `../../.cache/${buildtype}/drupal/pages.json`),
+  ),
+);
+
+const assembleEntityTree = assembleEntityTreeFactory(exportDir);
+
+if (help) {
```

Comment:
Should we move the `help` check before the `JSON.parse` and tree assembly? If I'm reading it correctly, it seems like 1. adding the `help` arg would still parse JSON (which seems unexpected), and 2. the `help` command could fail if there was an error during the parse or assembly. 

Does my understanding makes sense, or am I missing something?

Reply:
That's a good point. I just pushed a commit that moved the `help` check up.